config/environment.js

```diff
@@ -0,0 +1,17 @@
+'use strict';
+
+module.exports = function(environment) {
+  var results = {
+    APP: {
```

Comment:
[Relevant code.](https://github.com/ember-cli/ember-cli/blob/20323eee424f27cd0217b21ffd05a59f3dc2aa59/lib/models/project.js#L187-L229)

[It's a deep merge.](https://lodash.com/docs#merge)